### PR TITLE
Regenerate with gtk 3.99.3

### PR DESCRIPTION
**Pull request by bilelmoussaoui**
Depends on #35

### CODE REVIEW

**Comment by sdroege** [comment]
You seem to have forgotten to create the `gir-files` PR :)

File: gdk4/src/auto/toplevel.rs [hunk 1]
```diff
@@ -50,15 +49,15 @@ pub trait ToplevelExt: 'static {
 
     fn focus(&self, timestamp: u32);
 
-    fn get_state(&self) -> SurfaceState;
+    //fn get_state(&self) -> /*Ignored*/ToplevelState;
```
Comment (sdroege) [inline gdk4/src/auto/toplevel.rs:52]: This should be easy to generate by simply putting the enum into the generate list :)

File: gtk4/src/auto/functions.rs [hunk 2]
```diff
@@ -138,10 +137,9 @@ pub fn disable_setlocale() {
 //    unsafe { TODO: call gtk_sys:gtk_distribute_natural_allocation() }
 //}
 
-pub fn get_debug_flags() -> u32 {
-    assert_initialized_main_thread!();
-    unsafe { gtk_sys::gtk_get_debug_flags() }
-}
+//pub fn get_debug_flags() -> /*Ignored*/DebugFlags {
```
Comment (sdroege) [inline gtk4/src/auto/functions.rs:140]: Also an easy enum :)

File: gtk4/src/auto/gesture.rs [hunk 3]
```diff
@@ -68,20 +68,32 @@ pub trait GestureExt: 'static {
 
     fn get_property_n_points(&self) -> u32;
 
-    fn connect_begin<F: Fn(&Self, &gdk::EventSequence) + 'static>(&self, f: F) -> SignalHandlerId;
+    fn connect_begin<F: Fn(&Self, Option<&gdk::EventSequence>) + 'static>(
```
Comment (sdroege) [inline gtk4/src/auto/gesture.rs:71]: This and below fixes https://github.com/gtk-rs/gtk4/issues/29

File: gtk4/src/auto/tree_store.rs [hunk 4]
```diff
@@ -60,7 +60,7 @@ pub trait TreeStoreExt: 'static {
 
     fn prepend(&self, parent: Option<&TreeIter>) -> TreeIter;
 
-    fn remove(&self, iter: &TreeIter) -> bool;
+    //fn remove(&self, iter: /*Unimplemented*/TreeIter) -> bool;
```
Comment (sdroege) [inline gtk4/src/auto/tree_store.rs:63]: Hm why? `TreeIter` bindings exist

**Review by sdroege — COMMENTED** [review]
Looks generally good to me apart from those comments.

File: gtk4/src/auto/tree_store.rs (hunk 4 above)
Comment (bilelmoussaoui) [reply gtk4/src/auto/tree_store.rs:63]: Hmm no idea, maybe the annotation is broken on the latest release ? I will check

Comment (sdroege) [reply gtk4/src/auto/tree_store.rs:63]: Something seems clearly broken, yes. Two lines above the other function works fine :)

File: gtk4/src/auto/gesture.rs (hunk 3 above)
Comment (sdroege) [reply gtk4/src/auto/gesture.rs:71]: What I mean here: add a `Fixes https://github.com/gtk-rs/gtk4/issues/29` to the commit message :)

**Comment by bilelmoussaoui** [comment]
> You seem to have forgotten to create the `gir-files` PR :)

This was handled by Chris few days ago already :)

**Comment by sdroege** [comment]
> This was handled by Chris few days ago already :)

https://github.com/gtk-rs/gir-files/pull/61 this one? And the bindings were simply not yet updated to that version?

**Comment by bilelmoussaoui** [comment]
Yes, that's it. There was an sdk update recently though, I will check if there were any changes as I need to add the necessary gir files for #37 as well

**Comment by sdroege** [comment]
Makes sense. Then get this in here once the CI is happy, the changes look good to me :)

**Review by sdroege — APPROVED** [review]

File: gtk4/src/auto/tree_store.rs (hunk 4 above)
Comment (bilelmoussaoui) [reply gtk4/src/auto/tree_store.rs:63]: Ah it uses inout
```xml
      <method name="remove" c:identifier="gtk_tree_store_remove">
        <doc xml:space="preserve">Removes @iter from @tree_store.  After being removed, @iter is set to the
next valid row at that level, or invalidated if it previously pointed to the
last one.</doc>
        
        <return-value transfer-ownership="none">
          <doc xml:space="preserve">%TRUE if @iter is still valid, %FALSE if not.</doc>
          <type name="gboolean" c:type="gboolean"/>
        </return-value>
        <parameters>
          <instance-parameter name="tree_store" transfer-ownership="none">
            <doc xml:space="preserve">A #GtkTreeStore</doc>
            <type name="TreeStore" c:type="GtkTreeStore*"/>
          </instance-parameter>
          <parameter name="iter" direction="inout" caller-allocates="0" transfer-ownership="full">
            <doc xml:space="preserve">A valid #GtkTreeIter</doc>
            <type name="TreeIter" c:type="GtkTreeIter*"/>
          </parameter>
        </parameters>
      </method>
```

**Comment by bilelmoussaoui** [comment]
This looks good now :)

File: gtk4/src/auto/tree_store.rs (hunk 4 above)
Comment (sdroege) [reply gtk4/src/auto/tree_store.rs:63]: Needs to be manually implemented for now then

Comment (bilelmoussaoui) [reply gtk4/src/auto/tree_store.rs:63]: I've opened https://github.com/gtk-rs/gtk4-rs/issues/40 to keep track of that

**Pull request merged**